components/common-go/namegen/workspaceid.go

```diff
@@ -15,13 +15,9 @@ import (
 
 // WorkspaceIDPattern is the expected Worksapce ID pattern
 // gitpod-protocol/src/util/generate-workspace-id.ts is authoritative over the generation
+// ws-proxy/pkg/proxy/workspacerouter.go is authoritative for this regexp
 
-/*
-  - Regex Pattern Description:
-    Repo. Owner & Repo. Name  	Random
-    3-23 chars  				11 chars
-*/
-var WorkspaceIDPattern = regexp.MustCompile(`^[a-z0-9-]{3,23}-[a-z0-9]{11}$`)
+var WorkspaceIDPattern = regexp.MustCompile(`^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$|^[0-9a-z]{2,16}-[0-9a-z]{2,16}-[0-9a-z]{8,11}$`)
```

Comment:
I wonder should not we have a shared regex pattern in common go library. ws-proxy has own definition here: https://github.com/gitpod-io/gitpod/blob/e7f269d127031b32bb7247f6adb2b304a67d122d/components/ws-proxy/pkg/proxy/workspacerouter.go#L29 They should be aligned and ws-proxy is very well tested already. cc @easyCZ @mustard-mh

Reply:
Yep, ideally we should.

Reply:
@mustard-mh if you would have time to send a clean up PR 🙏